crates/hir_ty/src/diagnostics/expr.rs

```diff
@@ -150,20 +154,54 @@ impl<'a, 'b> ExprValidator<'a, 'b> {
         }
     }
 
-    fn validate_call(&mut self, db: &dyn HirDatabase, call_id: ExprId, expr: &Expr) -> Option<()> {
+    fn check_for_filter_map_next(&mut self, db: &dyn HirDatabase) {
+        let body = db.body(self.owner.into());
+        let mut prev = None;
+
+        for (id, expr) in body.exprs.iter() {
+            if let Expr::MethodCall { receiver, method_name, args, .. } = expr {
+                let method_name = format!("{}", method_name);
```

Comment:
What should I do doing instead here?

Reply:
`method_name.to_string()` and `method_name == Name::new_text(SmolStr::new_inline("filter_map"))`, but there's a list of known names in `crates/hir_expand/src/name.rs` where you can add them, but I can't find where most of the interesting ones are used (e.g. `Iterator` or `Try`).

You could also check whether it's `Iterator::filter_map`, there's some code calling `method_resolution` below, though I'm not sure how to do this check.

Reply:
Thanks. I'll switch to `to_string()` if allocating a new string and doing a comparison is acceptable here. If not then I'll work out how to get the method resolution to work.

Reply:
I'd add it to `known` instead, to be consistent.